src/compact.rs

```diff
@@ -46,8 +46,14 @@ struct PrefixInput<'a, T> {
 }
 
 impl<'a, T: 'a + Input> Input for PrefixInput<'a, T> {
-	fn remaining_len(&mut self) -> Result<usize, Error> {
-		Ok(self.input.remaining_len()?.saturating_add(self.prefix.iter().count()))
+	fn remaining_len(&mut self) -> Result<Option<usize>, Error> {
+		let len = if let Some(len) = self.input.remaining_len()? {
```

Comment:
```suggestion
		let len = self.input.remaining_len()?.unwrap_or(0);
		Ok(Some(len.saturating_add(self.prefix.iter().count()))
```

Reply:
hmm not really if input returns none then returning 0 Some(0) or Some(1) is wrong. indeed we don't use it in compact but still.